project.mml

```diff
@@ -285,14 +285,22 @@ Layer:
       <<: *osm2pgsql
       table: |-
         (SELECT
-             way, "natural"
-           FROM planet_osm_polygon
-           WHERE "natural" IN ('spring')
-         UNION ALL
-         SELECT
-             way, "natural"
-           FROM planet_osm_point
-           WHERE "natural" IN ('spring')
+            way,
+            "natural"
+          FROM
+          (SELECT
+              ST_PointOnSurface(way) AS way,
```

Comment:
Question: is psql smart enough to only do the ST_PointOnSurface calculation for objects satisfying `"natural" IN ('spring')`, in this query?

Reply:
Ok, I see now it avoids the duplication of `WHERE "natural" IN ('spring')`.

Reply:
In *this* case the conditions on the tag columns are simple enough a subselect isn't necessary, but on longer queries, these can be over a dozen lines.

Reply:
Postgres can do the query either way, the results are the same. It will pick whichever is cheapest, which will normally be using the index, filtering by geometry, then calculating st_pointonsurface.

Reply:
So if I understand you correctly, in this query it will also calculate ST_PointOnSurface for polygons that are not natural=spring, right?

Reply:
There is no guarantee one way or the other. It's unlikely to, but with a small database and the right indexes it might.